HTML/HTML.sublime-syntax

```diff
@@ -310,6 +310,18 @@ contexts:
           embed_scope: source.js.embedded.html
           escape: (?i)(?=(?:-->\s*)?</script)
 
+  script-html:
+    - meta_content_scope: meta.tag.script.begin.html
+    - include: script-common
+    - match: '>'
+      scope: punctuation.definition.tag.end.html
+      set:
+        - include: script-close-tag
+        - match: (?=\S)
+          embed: scope:text.html.basic
+          embed_scope: text.html.embedded.html
+          escape: (?i)(?=(?:-->\s*)?</script)
```

Comment:
Do we need the close comment in the lookahead?

Reply:
Myself, I don't think we should have it at all.

It's there for code written to support *extremely* old browsers that didn't know about the `<script>` tag. It's not valid, and I can count on one hand the number times I've seen comments used like that. I'd just as soon change all of the `escape` lookaheads to `(?i)</script)` (which is what every browser from the past twenty years will actually look for) and remove the comment handling in `script-close-tag`.

Reply:
I was thinking in this situation, since this is a lookahead, and the embedded content is also HTML, the `-->` can be handled by the embedded HTML instead of being part of the lookahead.

Does that make sense?

Reply:
In other situations (such as JavaScript) we need to pop before the close comment, so it isn't detected as JS.

Reply:
1. Why does a language embed itself? 
2. What happens if the embedded html contains another `<script>...</script>` section? I guess it will trigger the `escape` and break highlighting?

Reply:
I was also wondering, why not just scope the `script` tag and not bother with the embed action if the type is `text/html`, but I guess the closing script tag would then need special handling because otherwise nothing would scope it correctly?

Reply:
>What happens if the embedded html contains another `<script>...</script>` section? I guess it will trigger the `escape` and break highlighting?

I guess this largely depends on how the templating engine is implemented. The HTML5 spec isn't exactly clear on that (to me):
https://www.w3.org/TR/html50/scripting-1.html#restrictions-for-contents-of-script-elements

Reply:
After thoroughly reviewing the canonical HTML5, parsing algorithm, I have reached the conclusion that it's a mess and my head hurts.

It won't be possible to exactly implement the correct algorithm; we'd have to maintain two stacks.

Reply:
> I guess this largely depends on how the templating engine is implemented

As the `escape` pattern is matched before any of the embedded rules the underlying implementation does not influence it in any way, I think. The very first `</script>` will cause the embedded syntax to be popped off from stack, without any chance for it to avoid that.

Just replace the added test by the following lines to see the new test to fail upon the embedded `</script>` tag.

```
        <script
        type
        =
        text/html>
            <div></div>
        ##  ^^^^^^^^^^^ text.html.embedded
            <script type=text/javascript>
        ##  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ text.html.embedded.html
                function test() {}
        ##      ^^^^^^^^^^^^^^^^^^ source.js.embedded.html
            </script>
        ##  ^^^^^^^^^ text.html.embedded.html
        </script>
```